subsys/testsuite/ztest/include/ztest_test_new.h

```diff
@@ -344,6 +361,45 @@ extern struct k_mem_partition ztest_mem_partition;
  */
 #define ztest_run_test_suite(suite) z_ztest_run_test_suite(STRINGIFY(suite))
 
+/**
+ * @brief Helper function to get command line argument for listing tests
+ *
+ * @return true
+ * @return false
+ */
+bool z_ztest_get_list_test(void);
+
+/**
+ * @brief Lists registered unit tests in this binary
+ *
+ * @return int Number of tests in binary
+ */
+int z_ztest_list_tests(void);
```

Comment:
I'm still not convinced that we need the `list` feature here. It forces us to expose a lot of internal functions. This should be done by a separate python script in my opinion. It should be simple enough to find the sections in the `.elf` file and generate the list from there.

Reply:
> This should be done by a separate python script in my opinion. It should be simple enough to find the sections in the .elf file and generate the list from there.

This sounds needlessly complex compared to what @asemjonovs has here. What justifies this extra complexity?

Reply:
Not requiring the implementation of `z_test_get_list_test()` for every architecture where it's really only needed for 1.

How about this... Can we just split the implementation of `ztest_main_default()`? We can have one version of it in `ztest_defaults.c` and one in `ztest_posix.c`. Then only the posix one will include the functionality for listing.

Reply:
Agree, this approach is cleaner. Updated review with these changes.